pkg/motor/query_bucket.go

```diff
@@ -34,3 +36,51 @@ func (mtr *motorNodeImpl) QueryWhereIsByCreator(ctx context.Context) (mt.QueryWh
 		WhereIs: resp,
 	}, nil
 }
+
+func (mtr *motorNodeImpl) QueryBucketBySchema(ctx context.Context, req mt.QueryBucketContentBySchemaRequest) (mt.QueryBucketContentBySchemaResponse, error) {
+	if req.Creator != mtr.Address {
+		return mt.QueryBucketContentBySchemaResponse{}, fmt.Errorf("creator address does not match session creator: %s", mtr.Address)
+	}
```

Comment:
Do we need this?

Reply:
+1 You should be able to query schemas you didn't create.

This feature in general will be improved when we add namespacing

Reply:
```
if b.WhereIs.Visibility == types.BucketVisibility_PRIVATE && b.WhereIs.Creator != mtr.Address {
	return mt.QueryBucketContentBySchemaResponse{}, fmt.Errorf("creator address does not match session creator: %s", mtr.Address)
}`
```